dali/base/dautils.hpp

```diff
@@ -418,4 +419,40 @@ interface ILocalOrDistributedFile: extends IInterface
 
 extern da_decl ILocalOrDistributedFile* createLocalOrDistributedFile(const char *fname,IUserDescriptor *user,bool onlylocal,bool onlydfs,bool iswrite=false);
 
+struct LockData
+{
+    unsigned mode;
+    SessionId sessId;
+    unsigned timeLockObtained;
+};
+
+typedef __int64 ConnectionId;
+typedef MapBetween<ConnectionId, ConnectionId, LockData, LockData> ConnectionInfoMap;
+
+enum LockDataField
+{
+    LDFXPath = 0,
+    LDFEPCount = 1,
+    LDFEP = 2,
+    LDFconnId = 3,
+    LDFsessId = 4,
+    LDFmode = 5,
+    LDFtimeLocked = 6
+};
+
+class CLockDataHelper
+{
+    bool checkEP(const char *ep, const char *ipPattern);
+    void formatLock(IPropertyTree &query, bool fileLock, unsigned formatType, bool &headerDone, const char *ipPattern, const char *xpathPattern,
+        Int64Array *connIds, StringBuffer &out);
+
+public:
+    CLockDataHelper() {};
+    void serializeLockData(CheckedCriticalSection &crit, unsigned critTimeout, const char *xpath, ConnectionInfoMap &connectionInfo, CMessageBuffer &lockInfo);
+    IPropertyTreeIterator *getLockDataTreeIterator(MemoryBuffer &lockInfo);
+    StringBuffer &formatLockData(MemoryBuffer &lockInfo, StringBuffer &out);
```

Comment:
NB: should probably be a static, don't need an object to call this formatting conversion.